app/services/base_api_manager.rb

```diff
@@ -7,28 +7,28 @@ def initialize(organization = "default")
   end
 
   #Fetches all the resources for the specific entity
-  def get_entities(entity_name = "", opts={})
+  def get_entities(entity_name="", opts={}, last_synchronization_date=nil, after_last_sync=false)
     begin
       batched_call = opts[:__skip] && opts[:__limit]
-      #the page requested is calculated from the offset ([:__skip)]) and (opts[:__limit])
-      if batched_call
-        page = opts[:__skip] == 0 ? 1 : (opts[:__skip] / opts[:__limit] + 1)
-        query = "page=#{page}&per_page=#{opts[:__limit]}"
-      end
+
+      query = QueryParamsManager.batched_call(opts) if batched_call
+      query = QueryParamsManager.by_updated_at if after_last_sync
+
       response = RestClient.get "https://api.getbase.com/v2/#{entity_name.downcase.pluralize}?#{query}", headers_get
       #the meta field is retrieved indipendently to avoid conflicting with DataParser
       meta = JSON.parse(response)['meta']
       #DataParser strips the response from 'items' and 'data' fields
-      entities = DataParser.from_base_collection(response)
+      entities = DataParser.from_base_collection(response.body)
+
 
       unless batched_call
         while meta['links']['next_page']
-          response = Restclient.get "#{meta['links']['next_page']}", headers_get
-          raise 'No response received while fetching subsequent page' unless response
-          entities << DataParser.from_base_collection(response)
+          response = RestClient.get "#{meta['links']['next_page']}", headers_get
+          entities.concat DataParser.from_base_collection(response.body)
+          raise 'No response received while fetching subsequent page' unless response && !response.body.blank?
+          break if after_last_sync && Time.parse(entities.last['updated_at']) < last_synchronization_date
```

Comment:
I'm not sure you need the Time.parse, rails does it on its own when comparing a string to a Time